From 05d08228456566a5dcc8502b4801cb85f31f4fdb Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:01:01 -0400
Subject: [fs] btrfs: send: remove BUG from process_all_refs

Message-id: <1407179109-13714-175-git-send-email-sandeen@redhat.com>
Patchwork-id: 87031
O-Subject: [RHEL7.1 PATCH 174/422] btrfs: send: remove BUG from process_all_refs
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: David Sterba <dsterba@suse.cz>

commit 4d1a63b21b4f77a82efe7d78fc1ae1cc7532691c
Author: David Sterba <dsterba@suse.cz>
Date:   Mon Feb 3 19:24:19 2014 +0100

    btrfs: send: remove BUG from process_all_refs

    There are only 2 static callers, the BUG would normally be never
    reached, but let's be nice.

    Signed-off-by: David Sterba <dsterba@suse.cz>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 4405aae..d3ed9df 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -3606,7 +3606,10 @@ static int process_all_refs(struct send_ctx *sctx,
   root = sctx->parent_root;
   cb = __record_deleted_ref;
  } else {
-  BUG();
+  btrfs_err(sctx->send_root->fs_info,
+    "Wrong command %d in process_all_refs", cmd);
+  ret = -EINVAL;
+  goto out;
  }
 
  key.objectid = sctx->cmp_key->objectid;
-- 
1.7.1